cmd/api/src/bootstrap/server.go

```diff
@@ -50,31 +42,27 @@ const (
 	ContentSecurityPolicy        = "default-src 'self'; script-src 'self' 'unsafe-inline'; style-src 'self' 'unsafe-inline'; img-src 'self' data: blob:; font-src 'self' data:;"
 )
 
-// SystemSignalExitChannel is used to shut down the server. It creates a channel that listens for an exit signal from the server.
-func SystemSignalExitChannel() chan struct{} {
-	exitC := make(chan struct{})
+func NewDaemonContext(parentCtx context.Context) context.Context {
+	daemonContext, doneFunc := context.WithCancel(parentCtx)
 
 	go func() {
+		defer doneFunc()
+
 		// Shutdown on SIGINT/SIGTERM
 		signalChannel := make(chan os.Signal, 1)
 		signal.Notify(signalChannel, syscall.SIGTERM)
 		signal.Notify(signalChannel, syscall.SIGINT)
 
 		// Wait for a signal from the OS
 		<-signalChannel
-		close(exitC)
 	}()
 
-	return exitC
+	return daemonContext
 }
 
 // MigrateGraph runs migrations for the graph database
-func MigrateGraph(cfg config.Configuration, db graph.Database) error {
-	if cfg.DisableMigrations {
-		log.Infof("Graph migrations are disabled per configuration")
-		return nil
-	}
-	return migrations.NewGraphMigrator(db).Migrate()
+func MigrateGraph(ctx context.Context, db graph.Database, schema graph.Schema) error {
+	return migrations.NewGraphMigrator(db).Migrate(ctx, schema)
```

Comment:
Consider hoisting the check for disabling migrations for better separation of concerns.

Reply:
Hoisted to entrypoints.